app/src/components/auth/auth.tsx

```diff
@@ -4,11 +4,26 @@ import {
   getAppEndpointKey,
   getStorageClientKey,
   getStorageNodeAuthorized,
+  setExecutorPublicKey,
 } from '../../utils/storage';
+import apiClient from '../../api';
 
 export default function WithIdAuth({ children }: any) {
   const router = useRouter();
 
+  useEffect(() => {
+    const setContextIdentity = async () => {
+      const response = await apiClient.node().getContext();
+      if (
+        response?.data?.contextIdentities &&
+        response.data.contextIdentities.length > 0
+      ) {
+        setExecutorPublicKey(response.data.contextIdentities[0]);
```

Comment:
This is potentially dangerous. But ok for now as we don't provide selector. When we will have selector this will need to change